### PR TITLE
boostification of all websocket signals

### PR DESCRIPTION
**Pull request by sima-fastly**
boostification of all websocket signals

tested manually

### CODE REVIEW

File: src/cpp/proxy/wsproxysession.cpp
Comment (jkarneges) [inline src/cpp/proxy/wsproxysession.cpp:1]: hmm, indentation changed in the whole file causing a huge diff

Comment (sima-fastly) [reply src/cpp/proxy/wsproxysession.cpp:1]: sheesh! my git client doesnt show the diff when the tabs turn into space and I keep missing em prior to PR.
Ill clean this up and resubmit

Comment (sima-fastly) [reply src/cpp/proxy/wsproxysession.cpp:1]: I also need to update my VS code indentation settings so it doesn't convert on save or sth

File: src/cpp/proxy/websocketoverhttp.cpp [hunk 1]
```diff
@@ -46,6 +46,12 @@
 
 namespace {
 
+struct WSConnections {
+    Connection disconnectedConnection;
+    Connection closedConnection;
+    Connection errorConnection;
+};
+
```
Comment (jkarneges) [inline src/cpp/proxy/websocketoverhttp.cpp:54]: indentation

File: src/cpp/proxy/wsproxysession.cpp [hunk 2]
```diff
@@ -291,6 +309,8 @@ class WsProxySession::Private : public QObject
 	Connection keepAliveConneciton;
 	Connection aboutToSendRequestConnection;
 	map<WsControlSession*, WSProxyConnections> wsProxyConnectionMap;
+	map<WebSocket*, WSConnections> outWSConnectionMap;
+	map<WebSocket*, InWSConnections> inWSConnectionMap;
```
Comment (jkarneges) [inline src/cpp/proxy/wsproxysession.cpp:313]: There is only ever one in sock and one out sock.

File: src/cpp/proxy/sockjssession.cpp [hunk 3]
```diff
@@ -244,7 +244,6 @@ class SockJsSession::Private : public QObject
 		}
 		requests.clear();
 
-		wsConnectionMap.erase(sock);
```
Comment (jkarneges) [inline src/cpp/proxy/sockjssession.cpp]: should probably still disconnect here somehow?

the connections object seems to have a default constructor, so maybe just:
```
wsConnection = WSConnections();
```

File: src/cpp/proxy/wsproxysession.cpp [hunk 4]
```diff
@@ -352,7 +352,6 @@ class WsProxySession::Private : public QObject
 
 		cleanupInSock();
 
-		outWSConnectionMap.erase(outSock);
```
Comment (jkarneges) [inline src/cpp/proxy/wsproxysession.cpp]: same (and others)

File: src/cpp/proxy/wsproxysession.cpp [hunk 5]
```diff
@@ -372,7 +371,6 @@ class WsProxySession::Private : public QObject
 		if(inSock)
 		{
 			connectionManager->removeConnection(inSock);
-			inWSConnectionMap.erase(inSock);
```
Comment (jkarneges) [inline src/cpp/proxy/wsproxysession.cpp]: same

**Review by sima-fastly — COMMENTED** [review]
@jkarneges I have updated the code to delete connection struct instace when a req/sock gets deleted.

but this reminds me there are other places such as this file : 
https://github.com/fastly/pushpin/blob/adf96ae5692c8c45e6835b40ea4dfb51f588142f/src/cpp/proxy/requestsession.cpp#L238C4-L238C23
where a req or socket are getting deleted without the corresponding connection getting disconnected. Am I correct in thinking these also need a disconnect statement fix?

**Comment by jkarneges** [comment]
Ah yeah, anywhere an object is deleted that might have had connections, especially if the deletion is done from somewhere that isn't the destructor and the object pointer is also nulled out, any connections should also be disconnected.

I suggest keeping the connections object as a value and not a pointer, or else you'll need to null out the connections object too. The connection object is initially implicitly constructed using its default constructor, so explicitly calling the default constructor later on should return it to its initial state.

**Comment by sima-fastly** [comment]
> Ah yeah, anywhere an object is deleted that might have had connections, especially if the deletion is done from somewhere that isn't the destructor and the object pointer is also nulled out, any connections should also be disconnected.
> 
> I suggest keeping the connections object as a value and not a pointer, or else you'll need to null out the connections object too. The connection object is initially implicitly constructed using its default constructor, so explicitly calling the default constructor later on should return it to its initial state.

I have addressed this and will have a follow up PR to clean these up in other files

**Review by jkarneges — APPROVED** [review]

**Pull request merged**